### PR TITLE
fix: warn user when i18n weekday list size is not 7

### LINKED ISSUE #6353: Weekdays in DatePicker with I18N are not aligned
### Description

If I configure I18n in a DatePicker, the weekdays are not stretched. For instance, in the following image (left), "Sun" seems to be displayed above July 1. Compare to a DatePicker instance without further configuration (right)
image

### Expected outcome

I would expect both DatePickers to look similar.

### Minimal reproducible example

```
    DatePicker field1 = new DatePicker();
    DatePicker field2 = new DatePicker();

    DatePickerI18n i18n = new DatePickerI18n();
    DateFormatSymbols dfs = DateFormatSymbols.getInstance(Locale.ENGLISH);
    i18n.setMonthNames(List.of(dfs.getMonths()));
    i18n.setWeekdays(List.of(dfs.getWeekdays()));
    i18n.setWeekdaysShort(List.of(dfs.getShortWeekdays()));
    add(field1, field2);
    field1.setI18n(i18n);
```

### Steps to reproduce

1. Add the snippet above to a view. 
2. Open each `DatePicker` popup. 
3. Notice how "Sun" seems to be displayed above July 1.

### Environment

Vaadin version(s):  24.3.13
Browser: Chrome 125

### Browsers

_No response_

### PR DESCRIPTION
**Pull request by FrediWa**
## Description

`setWeekdays` should always get 7 elements. Notify the user/developer if they are setting a list with length not equal to 7.

Fixes #6353 
Related PR: https://github.com/vaadin/web-components/pull/7542

### CODE REVIEW

File: vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import org.slf4j.LoggerFactory;
```
Comment (sissbruecker) [inline vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:30]: Nit: Order of this import could be aligned with how the other imports are structured.

File: vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java [hunk 2]
```diff
@@ -899,6 +901,10 @@ public List<String> getWeekdays() {
          * @return this instance for method chaining
          */
         public DatePickerI18n setWeekdays(List<String> weekdays) {
+            if (weekdays != null && weekdays.size() != 7) {
+                LoggerFactory.getLogger(getClass().getName()).warn(String.format("setWeekdays input list should have exactly 7 elements. Instead got %s", weekdays.size()));
```
Comment (sissbruecker) [inline vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:905]: - `getClass` is enough to get the logger
- "parameter list" might be more appropriate to reference the method parameter than "input list"
- Not sure how you bypassed the formatting check in the CI, it should have complained 😅. Anyway, before committing your code, try to remember to run `mvn formatter:format`.

File: vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java [hunk 3]
```diff
@@ -921,6 +927,10 @@ public List<String> getWeekdaysShort() {
          * @return this instance for method chaining
          */
         public DatePickerI18n setWeekdaysShort(List<String> weekdaysShort) {
+            if (weekdaysShort != null && weekdaysShort.size() != 7) {
+                LoggerFactory.getLogger(getClass().getName()).warn(String.format("setWeekdaysShort input list should have exactly 7 elements. Instead got %s", weekdaysShort.size()));
```
Comment (sissbruecker) [inline vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:931]: Same here.

File: vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java (hunk 1 above)
Comment (FrediWa) [reply vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:30]: Seems like this particular import is all over the place. In some cases it's before vaadin component imports, in others it's after. If it's important, should we do a PR that specifically updates the order of it?

Comment (sissbruecker) [reply vaadin-date-picker-flow-parent/vaadin-date-picker-flow/src/main/java/com/vaadin/flow/component/datepicker/DatePicker.java:30]: Not really important, let's leave as is. The imports in the codebase are basically ordered by whatever the respective developers IDE thinks is best. 

If we want to do something we should maybe have an automated way of ordering them, similar to the code formatter. Like in `web-components` for example where a linter that checks order of imports and can fix them if needed.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**